### PR TITLE
fix: Create Shared SBOM Workflow For AMI

### PR DESCRIPTION
**Pull request by Efe-Oddball**
### **Ticket**
https://github.com/orgs/department-of-veterans-affairs/projects/1283/views/1?pane=issue&itemId=69048726

### **Technical Summary**
Considering the existing shared SBOM workflow is for container images and trying to avoid putting alot of conditional logic within that workflow to accommodate SBOM creation for AMI, I had to create a separate workflow that will be used specifically for AMIs.
This takes in the AMI_ID as the artifact_identifier and passes it into the Trivy scan which has been set up with a cyclonedx format.

### **Test**
This has been successfully tested with the packer-hardened, packer-eks and ghar ami workflows.

### CODE REVIEW

**Review by ph-One — APPROVED** [review]

File: .github/workflows/sbom-generation.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ jobs:
       uses: aws-actions/amazon-ecr-login@v1
 
     - name: Run Trivy vulnerability scanner
-      uses: aquasecurity/trivy-action@0.17.0
+      uses: aquasecurity/trivy-action@0.19.0
```
Comment (ph-One) [inline .github/workflows/sbom-generation.yml:42]: see what the latest version is. I have seen 0.20, at a minimum

Comment (ph-One) [inline .github/workflows/sbom-generation.yml:42]: I know that we run 0.20 in other places. See what the latest release is

**Review by ph-One — CHANGES_REQUESTED** [review]

File: .github/workflows/sbom-generation.yml (hunk 1 above)
Comment (Efe-Oddball) [reply .github/workflows/sbom-generation.yml:42]: The latest release is 0.20. I will update that now

Comment (Efe-Oddball) [reply .github/workflows/sbom-generation.yml]: The latest release is 0.24.0. I will update these jobs

**Review by ph-One — APPROVED** [review]

**Pull request merged**